### PR TITLE
D-01242 - log source chain usage

### PR DESCRIPTION
**Pull request by robbiecarlton**
Counts the holochain database directories to calculate source chains, and logs it as part of the periodic `log_disk_usage` call.

Included in this holo-nixpkgs update: https://github.com/Holo-Host/holo-nixpkgs/pull/811

### CODE REVIEW

File: src/utils.ts [hunk 1]
```diff
@@ -1,9 +1,51 @@
+import { readdirSync } from 'fs'
+import { inspect } from 'util'
+import { groupBy, uniq } from 'lodash'
+
+const HOLOCHAIN_DATABASE_DIRECTORY = process.env.NODE_ENV === 'test'
+  ? './'
```
Comment (timotree3) [inline src/utils.ts:6]: This seems sketchy. Why do we want to use the current directory in tests? In an e2e test it seems like we would want to get the actual database directory.

**Review by timotree3 — CHANGES_REQUESTED** [review]
Looks good but I have a single question/concern. Also should there be envoy tests for this?

**Comment by alastairong** [comment]
Merging latest `develop` so that we can human-test for meeting AC

**Review by timotree3 — APPROVED** [review]
It sounds like we're not testing this feature anyway so I retract my comment in favor of getting this PR merged.

**Comment by alastairong** [comment]
Merged per discussion in stand up

**Pull request merged**